### PR TITLE
Stream load support auto __op mapping for json format

### PR DESCRIPTION
**Pull request by decster**

### CODE REVIEW

File: be/src/exec/vectorized/json_scanner.cpp [hunk 1]
```diff
@@ -398,11 +406,11 @@ void JsonReader::_construct_column(const rapidjson::Value& objectValue, Column*
         break;
     }
     case rapidjson::Type::kFalseType: {
-        column->append_strings(std::vector<Slice>{Slice("0")});
+        column->append_strings(literal_0_slice_vector);
         break;
     }
     case rapidjson::Type::kTrueType: {
-        column->append_strings(std::vector<Slice>{Slice("1")});
+        column->append_strings(literal_0_slice_vector);
```
Comment (Astralidea) [inline be/src/exec/vectorized/json_scanner.cpp:413]: literal_1_slice_vector is define and not to use.
maybe here should be literal_1_slice_vector?

Comment (decster) [reply be/src/exec/vectorized/json_scanner.cpp:413]: Yes, will fix

File: be/src/exec/vectorized/json_scanner.cpp [hunk 2]
```diff
@@ -26,6 +26,9 @@
 
 namespace starrocks::vectorized {
 
+static std::vector<Slice> literal_0_slice_vector{Slice("0")};
+static std::vector<Slice> literal_1_slice_vector{Slice("0")};
```
Comment (Linkerist) [inline be/src/exec/vectorized/json_scanner.cpp:30]: both literal_0_slice_vector and literal_1_slice_vector are Slice("0")?

File: be/src/exec/vectorized/json_scanner.cpp [hunk 3]
```diff
@@ -292,7 +295,12 @@ Status JsonReader::read_chunk(Chunk* chunk, int32_t rows_to_read, const std::vec
                     ColumnPtr& column = chunk->get_column_by_slot_id(slot_desc->id());
                     const char* column_name = slot_desc->col_name().c_str();
                     if (!objectValue->IsObject() || !objectValue->HasMember(column_name)) {
-                        column->append_nulls(1);
+                        if (strcmp(column_name, "__op") == 0) {
+                            // special treatment for __op column, fill default value '0' rather than null
+                            column->append_strings(literal_0_slice_vector);
+                        } else {
+                            column->append_nulls(1);
```
Comment (chaoyli) [inline be/src/exec/vectorized/json_scanner.cpp:302]: why not literal_1_slice_vector?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/exec/vectorized/json_scanner.cpp (hunk 3 above)
Comment (decster) [reply be/src/exec/vectorized/json_scanner.cpp:302]: cause 0 is the default value

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**